vllm/model_executor/layers/sampler.py

```diff
@@ -55,10 +57,14 @@ def forward(
         embedding_bias: Optional[torch.Tensor] = None,
     ) -> Optional[SamplerOutput]:
         # Get the hidden states that we use for sampling.
-        hidden_states = _prune_hidden_states(hidden_states, sampling_metadata)
+        if self.is_neuron:
```

Comment:
Please comment why we need an if over here.

Reply:
comment added above.

Reply:
It is because we generate logits directly in transformers-neuronx.